pkg/cache/internal/informers_map.go

```diff
@@ -34,45 +35,50 @@ import (
 	"k8s.io/client-go/metadata"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
-
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
 func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
-// clientListWatcherFunc knows how to create a ListWatcher.
-type createListWatcherFunc func(gvk schema.GroupVersionKind, ip *specificInformersMap) (*cache.ListWatch, error)
-
-// newSpecificInformersMap returns a new specificInformersMap (like
-// the generical InformersMap, except that it doesn't implement WaitForCacheSync).
-func newSpecificInformersMap(config *rest.Config,
-	scheme *runtime.Scheme,
-	mapper meta.RESTMapper,
-	resync time.Duration,
-	namespace string,
-	selectors SelectorsByGVK,
-	disableDeepCopy DisableDeepCopyByGVK,
-	transformers TransformFuncByObject,
-	createListWatcher createListWatcherFunc,
-) *specificInformersMap {
-	ip := &specificInformersMap{
-		config:            config,
-		Scheme:            scheme,
-		mapper:            mapper,
-		informersByGVK:    make(map[schema.GroupVersionKind]*MapEntry),
-		codecs:            serializer.NewCodecFactory(scheme),
-		paramCodec:        runtime.NewParameterCodec(scheme),
-		resync:            resync,
-		startWait:         make(chan struct{}),
-		createListWatcher: createListWatcher,
-		namespace:         namespace,
-		selectors:         selectors.forGVK,
-		disableDeepCopy:   disableDeepCopy,
-		transformers:      transformers,
+// InformersMapOptions configures an InformerMap.
+type InformersMapOptions struct {
+	Scheme       *runtime.Scheme
+	Mapper       meta.RESTMapper
+	ResyncPeriod time.Duration
+	Namespace    string
+	ByGVK        InformersMapOptionsByGVK
+}
+
+// InformersMapOptionsByGVK configured additional by group version kind (or object)
+// in an InformerMap.
+type InformersMapOptionsByGVK struct {
+	Selectors       SelectorsByGVK
+	Transformers    TransformFuncByObject
```

Comment:
No strong objection but it seems a bit odd to have "ByGVK" and then have `TransformFuncByObject`.

Looking at `TransformFuncByObject` it seems to essentially also use GVK
```go
	Set(runtime.Object, *runtime.Scheme, cache.TransformFunc) error
	Get(schema.GroupVersionKind) cache.TransformFunc
```

Get directly works with GVK and only Set takes an Object but only uses it to infer the GVK.

Wondering if we could/should rename `TransformFuncByObject` to `TransformFuncByGVK` especially as I would say the main functionality is mapping from "something" to `TransformFunc` in `Get` and that is done by GVK

Reply:
Renamed, it was all internal functions fortunately